coopcycle/coopcycle-web
# Feature: improve order validation

Fixes:

- https://github.com/coopcycle/coopcycle/issues/56
- https://github.com/coopcycle/coopcycle/issues/55

--

added: display validation errors on page load (address page & payment page)
added: validate order before processing the payment

added/updated: checkout end-to-end tests (includes: https://github.com/coopcycle/coopcycle-web/pull/4264 )

refactoring: move /checkout into /order (to match URL path)
refactoring: group payment-related components
refactoring: checkout log messages

--

How to test:

1. Create an order with a fixed time range (not ASAP)
2. Close a restaurant from the admin dashboard while you are on the different steps of the checkout flow (to simulate a situation when the time range is not valid any more)
3. See error message
